cli/packages/prisma-cli-core/src/commands/seed/Seeder.ts

```diff
@@ -97,13 +97,30 @@ export class Seeder {
     }
 
     const query = fs.readFileSync(filePath, 'utf-8')
+    let operations: string[] = []
     try {
-      parse(query)
+      const ast = parse(query)
+      operations = ast.definitions
+        .filter(d => d.kind === 'OperationDefinition')
+        .map(d => {
+          return print({
+            kind: 'Document',
+            definitions: [d],
+          })
+        })
     } catch (e) {
       throw new Error(`Error while parsing ${filePath}:\n${e.message}`)
     }
 
-    await this.client.exec(serviceName, stageName, query, token, workspaceSlug)
+    operations.forEach(async operation => {
```

Comment:
This can be added to the `try` and the `let operations` can then be turned into `const operations`, as this code wouldn't be reached anyways when the error would be thrown.

Reply:
In order to be able to wait for all these operations, they have to run with either `Promise.all` or `for of`

Reply:
Thanks! nice catch. This is now done :)